### ISSUE TITLE
bug: having a typo in map.txt can silently crash the game when firing weapons

### ISSUE THREAD

**Issue by EndrosG** [endless-sky/endless-sky#2438]
It took me some two hours? to find the source of my recent game crashes it in my data file :-/
I had two errors in my plugin/data/map.txt

```
system ...
[...]
	minables aluminium 2 2.757
[...]
	object
		sprite planet/desert10
		distance 547.187
		period 102.351
		object
			sprite planet/lava2
			distance 215rrr
			period 15.5389

```

The game loads without any notification or error.
I can land on the inhabited planets (the above one aint...), and can depart from them.
All looks good until I fire a weapon. In the moment I hit the key the game crashes with a memory access error (written to the console, no entry in error log).

When I found the `rrr` in the distance I thought like: yeah, i found it........ But the game there ignores the letters and just extracts the number. Smart move. :-)

The problem was the minables alumin**i**um. In my first language, we spell that with ium at the end and that was simply a typo of mine.
The game does not complain upon loading but simply crashes when you fire. Probably because it tries to find out if you hit something and cant check if you hit a wrong spelled minable.

**Comment by MessyMix** [comment]
I think your title is unclear for what you are saying. [I originally thought that there was a typo in maps.txt leading the game to crash, but that isn't the case. You are just saying that _if_ you have a typo, it will crash without notification.] So perhaps I would change it to something along the lines of: Having a typo in maps.txt leads to silent crash when firing weapons.

Side note: the official, scientifically accepted name for element 13 is aluminium with an -ium at the end, so technically you never made a typo.. :P

**Comment by endless-sky** [comment]
Fixed in 6d70bb9.

**Comment by EndrosG** [comment]
thx. :+1:
